### PR TITLE
ci: add build e2e test for swift

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

See https://github.com/aws-amplify/amplify-codegen/pull/513 for similar test for TypeScript apps.

#### Issue #, if available

PR to introduce the error: https://github.com/aws-amplify/amplify-codegen/pull/489

PR to fix the error: https://github.com/aws-amplify/amplify-codegen/pull/500

#### Description of how you validated changes

[Run on CircleCI](https://app.circleci.com/pipelines/github/aws-amplify/amplify-codegen/2643/workflows/92d3cc95-6956-4f8a-ae20-9235e61d940b)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/main/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-codegen-e2e-tests/src/__tests__/build-app-swift.test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  addApiWithBlankSchemaAndConflictDetection,
+  updateApiSchemaWithText,
+  generateModels,
+  swiftBuild,
+} from '@aws-amplify/amplify-codegen-e2e-core';
+const { schemas } = require('@aws-amplify/graphql-schema-test-library');
+import { existsSync, writeFileSync, readdirSync, rmSync, readFileSync } from 'fs';
+import path from 'path';
+
+const skip = new Set([
```
Comment (dpilch) [inline packages/amplify-codegen-e2e-tests/src/__tests__/build-app-swift.test.ts:13]: These cases are consistently failing. This indicates an existing issue with codegen for these cases.

File: .circleci/config.base.yml [hunk 2]
```diff
@@ -59,13 +69,13 @@ jobs:
       - checkout
       - run: yarn config set workspaces-experimental true
       - run: yarn cache clean --force
-      - run: yarn run production-build
+      - run: CXXFLAGS="--std=c++17" yarn run production-build
```
Comment (alharris-at) [inline .circleci/config.base.yml:72]: Do we want this for all runs?

File: packages/amplify-codegen-e2e-tests/test-apps/swift/swift/swiftApp.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+func configureAmplify() {
+    let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels())
+    do {
+        try Amplify.add(plugin: dataStorePlugin)
+        try Amplify.configure()
+        print("Initialized Amplify");
+    } catch {
+        // simplified error handling for the tutorial
+        print("Could not initialize Amplify: \(error)")
+    }
```
Comment (alharris-at) [inline packages/amplify-codegen-e2e-tests/test-apps/swift/swift/swiftApp.swift:33]: Should this just throw for the purposes of our test app?

Comment (alharris-at) [reply packages/amplify-codegen-e2e-tests/test-apps/swift/swift/swiftApp.swift:33]: Actually, I guess runtime errors wouldn't be caught in the current setup, right?

Comment (dpilch) [reply packages/amplify-codegen-e2e-tests/test-apps/swift/swift/swiftApp.swift:33]: Yeah, the test does not assert against runtime, only build time.

File: .circleci/config.base.yml (hunk 2 above)
Comment (dpilch) [reply .circleci/config.base.yml]: Set it for only the mac executor now.

**Review by alharris-at — APPROVED** [review]

**Review by david-mcafee — APPROVED** [review]
LGTM!

**Pull request merged**